app/Services/WebsiteUpdater/WebsiteUpdaterThroughGithubActions.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Services\WebsiteUpdater;
+
+use GuzzleHttp\ClientInterface;
+use RuntimeException;
+
+final class WebsiteUpdaterThroughGithubActions implements WebsiteUpdater
+{
+    public function __construct(
+        private ClientInterface $client,
+        private string $githubRepo,
+        private string $githubToken
+    ) {
+    }
```

Comment:
Is dit een magische constructor? Waar worden de attributes assigned? 🤔

Reply:
Dit is [property promotion](https://www.php.net/manual/en/language.oop5.decon.php#language.oop5.decon.constructor.promotion) dat is nieuw in php8. Hiermee kan je in 1 klap een field maken en de argument uit je constructor asignen, dit is een dingetje uit C# en typescript, erg fijn.